### ISSUE TITLE
crash in thermos server

### ISSUE THREAD

**Issue by Twilight2** [TheCrafter4000/TinkersMiddleearth#1]
Pick up the book will cause crash when backpack and hotbar are full of other items

**Comment by szczypikrul** [comment]
@Twilight2  did you find a solution?

**Comment by Twilight2** [comment]
> @Twilight2 did you find a solution?

Well...always put a book in player's backpack or even in a pouch.
It works until they put the book somewhere else or the stick side has reached 64,and everytime it happens,I need to solve the problem manully.
@szczypikrul

**Comment by TheCrafter4000** [comment]
I cannot reproduce this. What do you mean by "pick up the book when inventory is full"? Can you provide a more detailed description of what is happening and when it crashes? Can you provide a crash log?
@Twilight2 @szczypikrul

**Comment by Twilight2** [comment]
> 

I dont know why but each time when player login，the mod's manual will automatically give one to the player.And if player's inventory is full,this will cause crash.

I'll try to reproduce it today and recheck the logs.

**Comment by Twilight2** [comment]
[20:39:00] [Server thread/ERROR]: Exception caught during firing event net.minecraftforge.event.entity.player.EntityItemPickupEvent@31b29e72:
java.lang.ArrayIndexOutOfBoundsException: 0
	at mantle.items.abstracts.CraftingItem.func_77667_c(CraftingItem.java:70) ~[CraftingItem.class:?]
	at net.minecraft.item.ItemStack.func_77977_a(ItemStack.java:441) ~[add.class:?]
	at com.brandon3055.draconicevolution.common.achievements.Achievements.entityPickupEvent(Achievements.java:118) ~[Achievements.class:?]
	at cpw.mods.fml.common.eventhandler.ASMEventHandler_575_Achievements_entityPickupEvent_EntityItemPickupEvent.invoke(.dynamic) ~[?:?]
	at cpw.mods.fml.common.eventhandler.ASMEventHandler.invoke(ASMEventHandler.java:54) ~[ASMEventHandler.class:1.7.10-1614.57]
	at cpw.mods.fml.common.eventhandler.EventBus.post(EventBus.java:140) [EventBus.class:1.7.10-1614.57]
	at com.creativemd.itemphysic.physics.ServerPhysic.onCollideWithPlayer(ServerPhysic.java:488) [ServerPhysic.class:?]
	at com.creativemd.itemphysic.physics.ServerPhysic.onCollideWithPlayer(ServerPhysic.java:472) [ServerPhysic.class:?]
	at net.minecraft.entity.item.EntityItem.func_70100_b_(EntityItem.java) [xk.class:?]
	at net.minecraft.entity.player.EntityPlayer.func_71044_o(EntityPlayer.java:692) [yz.class:?]
	at net.minecraft.entity.player.EntityPlayer.func_70636_d(EntityPlayer.java:683) [yz.class:?]
	at net.minecraft.entity.EntityLivingBase.func_70071_h_(EntityLivingBase.java:1954) [sv.class:?]
	at net.minecraft.entity.player.EntityPlayer.func_70071_h_(EntityPlayer.java:315) [yz.class:?]
	at net.minecraft.entity.player.EntityPlayerMP.func_71127_g(EntityPlayerMP.java:399) [mw.class:?]
	at net.minecraft.network.NetHandlerPlayServer.func_147347_a(NetHandlerPlayServer.java:488) [nh.class:?]
	at net.minecraft.network.play.client.C03PacketPlayer.func_148833_a(C03PacketPlayer.java:36) [jd.class:?]
	at net.minecraft.network.play.client.C03PacketPlayer$C06PacketPlayerPosLook.func_148833_a(C03PacketPlayer.java:234) [jf.class:?]
	at net.minecraft.network.NetworkManager.func_74428_b(NetworkManager.java:245) [ej.class:?]
	at net.minecraft.network.NetworkSystem.func_151269_c(NetworkSystem.java:181) [nc.class:?]
	at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:1023) [MinecraftServer.class:?]
	at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:432) [lt.class:?]
	at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:841) [MinecraftServer.class:?]
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:693) [MinecraftServer.class:?]
	at java.lang.Thread.run(Unknown Source) [?:1.8.0_333]

**Comment by Twilight2** [comment]
[20:39:00] [Server thread/WARN]: Failed to handle packet for 
net.minecraft.util.ReportedException: Ticking player
	at net.minecraft.entity.player.EntityPlayerMP.func_71127_g(EntityPlayerMP.java:477) ~[mw.class:?]
	at net.minecraft.network.NetHandlerPlayServer.func_147347_a(NetHandlerPlayServer.java:488) ~[nh.class:?]
	at net.minecraft.network.play.client.C03PacketPlayer.func_148833_a(C03PacketPlayer.java:36) ~[jd.class:?]
	at net.minecraft.network.play.client.C03PacketPlayer$C06PacketPlayerPosLook.func_148833_a(C03PacketPlayer.java:234) ~[jf.class:?]
	at net.minecraft.network.NetworkManager.func_74428_b(NetworkManager.java:245) ~[ej.class:?]
	at net.minecraft.network.NetworkSystem.func_151269_c(NetworkSystem.java:181) [nc.class:?]
	at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:1023) [MinecraftServer.class:?]
	at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:432) [lt.class:?]
	at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:841) [MinecraftServer.class:?]
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:693) [MinecraftServer.class:?]
	at java.lang.Thread.run(Unknown Source) [?:1.8.0_333]
Caused by: java.lang.ArrayIndexOutOfBoundsException: 0
	at mantle.items.abstracts.CraftingItem.func_77667_c(CraftingItem.java:70) ~[CraftingItem.class:?]
	at net.minecraft.item.ItemStack.func_77977_a(ItemStack.java:441) ~[add.class:?]
	at com.brandon3055.draconicevolution.common.achievements.Achievements.entityPickupEvent(Achievements.java:118) ~[Achievements.class:?]
	at cpw.mods.fml.common.eventhandler.ASMEventHandler_575_Achievements_entityPickupEvent_EntityItemPickupEvent.invoke(.dynamic) ~[?:?]
	at cpw.mods.fml.common.eventhandler.ASMEventHandler.invoke(ASMEventHandler.java:54) ~[ASMEventHandler.class:1.7.10-1614.57]
	at cpw.mods.fml.common.eventhandler.EventBus.post(EventBus.java:140) ~[EventBus.class:1.7.10-1614.57]
	at com.creativemd.itemphysic.physics.ServerPhysic.onCollideWithPlayer(ServerPhysic.java:488) ~[ServerPhysic.class:?]
	at com.creativemd.itemphysic.physics.ServerPhysic.onCollideWithPlayer(ServerPhysic.java:472) ~[ServerPhysic.class:?]
	at net.minecraft.entity.item.EntityItem.func_70100_b_(EntityItem.java) ~[xk.class:?]
	at net.minecraft.entity.player.EntityPlayer.func_71044_o(EntityPlayer.java:692) ~[yz.class:?]
	at net.minecraft.entity.player.EntityPlayer.func_70636_d(EntityPlayer.java:683) ~[yz.class:?]
	at net.minecraft.entity.EntityLivingBase.func_70071_h_(EntityLivingBase.java:1954) ~[sv.class:?]
	at net.minecraft.entity.player.EntityPlayer.func_70071_h_(EntityPlayer.java:315) ~[yz.class:?]
	at net.minecraft.entity.player.EntityPlayerMP.func_71127_g(EntityPlayerMP.java:399) ~[mw.class:?]
	... 10 more
[20:39:00] [Server thread/INFO]: Twilight2 lost connection: Internal server error

**Comment by Twilight2** [comment]
These logs are on the serverside,nothing happens in client's log.
@TheCrafter4000

**Comment by TheCrafter4000** [comment]
> I dont know why but each time when player login，the mod's manual will automatically give one to the player.

This is not intended. It should only give one the first time a player joins, not every time. Would need more information to debug this, tho.

> And if player's inventory is full,this will cause crash.

Regarding the crash, I think I found the issue. Will publish a new version shortly.

**Comment by Twilight2** [comment]
Yes,it should give only once,but the playernbt maybe for the check or something,it reset everytime.
And even simply pick up the manual will cause crash too.
@TheCrafter4000

**Comment by TheCrafter4000** [comment]
> And if player's inventory is full,this will cause crash.

> And even simply pick up the manual will cause crash too.

Version 1.3.8 should fix this. Since I couldn't reproduce the exact same issue, please download it on [CurseForge](https://legacy.curseforge.com/minecraft/mc-mods/tinkersmiddleearth/files/4733750) and verify it works @Twilight2

> Yes,it should give only once,but the playernbt maybe for the check or something,it reset everytime.

If this is not also fixed by the update, please open a new issue for that and provide me a full server log

**Comment by Twilight2** [comment]
Yes,I just download the mod on curseforge and add it to my server,it still happens

**Comment by TheCrafter4000** [comment]
:(
Can you give me some logs? Either a crash log or full server log

**Comment by Twilight2** [comment]
[latest.log](https://github.com/TheCrafter4000/TinkersMiddleearth/files/12504347/latest.log)
@TheCrafter4000

**Comment by TheCrafter4000** [comment]
Are you sure you're using version 1.3.8? According to the logfile you are still using the old version 1.3.7b @Twilight2

**Comment by Twilight2** [comment]
Thanks！It fixed.
@TheCrafter4000
